src/wren_core.c

```diff
@@ -930,11 +941,73 @@ DEF_NATIVE(string_count)
   RETURN_NUM(count);
 }
 
+DEF_NATIVE(string_endsWith)
+{
+  if (!validateString(vm, args, 1, "Argument")) return PRIM_ERROR;
+
+  ObjString* string = AS_STRING(args[0]);
+  ObjString* search = AS_STRING(args[1]);
+
+  // Corner case, if the search string is longer than return false right away.
+  if (search->length > string->length) RETURN_FALSE;
+
+  int result = strncmp(string->value + string->length - search->length,
+                       search->value, search->length);
+
+  RETURN_BOOL(result == 0);
+}
+
+DEF_NATIVE(string_indexOf)
+{
+  if (!validateString(vm, args, 1, "Argument")) return PRIM_ERROR;
+
+  ObjString* string = AS_STRING(args[0]);
+  ObjString* search = AS_STRING(args[1]);
+
+  char* firstOccurrence = strstr(string->value, search->value);
+
+  RETURN_NUM(firstOccurrence ? (firstOccurrence - string->value) : -1);
+}
+
+DEF_NATIVE(string_startsWith)
+{
+  if (!validateString(vm, args, 1, "Argument")) return PRIM_ERROR;
+
+  ObjString* string = AS_STRING(args[0]);
+  ObjString* search = AS_STRING(args[1]);
+
+  RETURN_BOOL(strncmp(string->value, search->value, search->length) == 0);
```

Comment:
You could use `memcmp` here too. You also probably want to avoid the function call when the length of the search string is greater than the length of the string being searched, like you did in `endsWith`.